### PR TITLE
DAOS-8331 client: Telemetry dump should go to unique file paths

### PR DESCRIPTION
**Pull request by mjmac**
Rename D_CLIENT_METRICS_DUMP_PATH to D_CLIENT_METRICS_DUMP_DIR
and update the logic to create a unique file for each process
under that directory.

Required-githooks: true
Change-Id: If7854c9906fad213a12b94e09fc9974392f5bcde
Signed-off-by: Michael MacDonald <mjmac@google.com>

### CODE REVIEW

File: src/cart/README.env
Comment (mchaarawi) [inline src/cart/README.env:196]: as discussed in the other PR, imo _DIR is better/clearer than _PATH, but im not too hung up on that.
also the description needs to be updated.

Comment (mjmac) [reply src/cart/README.env]: Thanks. I probably should have kept this one in the draft state until the base was merged. It's ready to go now, though.

**Review by kjacque — DISMISSED** [review]
A couple minor suggestions but nothing I'd block on.

File: src/client/api/metrics.c
Comment (mjmac) [reply src/client/api/metrics.c]: Done.

File: src/client/api/metrics.c
Comment (mjmac) [reply src/client/api/metrics.c]: Thanks, that's better.

**Review by kjacque — APPROVED** [review]

**Comment by mjmac** [comment]
Failures appear to be unrelated:
  * [DAOS-15761](https://daosio.atlassian.net/browse/DAOS-15761): DFS stat mtime failure
  * [DAOS-15431](https://daosio.atlassian.net/browse/DAOS-15431): Rebuild simple timeout

**Comment by mjmac** [comment]
@mchaarawi: Nudge... This is waiting on a review from you, thanks in advance.

**Comment by mchaarawi** [comment]
> * [DAOS-15761](https://daosio.atlassian.net/browse/DAOS-15761): DFS stat mtime failure

i have not seen that failure before, so it's not a known issue since it's a new ticket you are creating as a result of this PR. that being said, the jenkins run here says it has passed:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14117/14/artifact/Functional%20Hardware%20Large/daos_test/dfs.py/
so not sure where that failure is coming from.

**Comment by mchaarawi** [comment]
> > * [DAOS-15761](https://daosio.atlassian.net/browse/DAOS-15761): DFS stat mtime failure
> 
> i have not seen that failure before, so it's not a known issue since it's a new ticket you are creating as a result of this PR. that being said, the jenkins run here says it has passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14117/14/artifact/Functional%20Hardware%20Large/daos_test/dfs.py/ so not sure where that failure is coming from.

i checked with phil and yes this is a cmocka failure of that test there. it has not been seen on master but only seen once on 1 other PR from Tom. so it's probably some intermittent regression introduced before.

**Review by mchaarawi — APPROVED** [review]

**Comment by mjmac** [comment]
@mchaarawi: Is there something more that needs to happen with this PR before it can be force-landed?

**Comment by mchaarawi** [comment]
> @mchaarawi: Is there something more that needs to happen with this PR before it can be force-landed?

hmm i did not see this in the gatekeeper queue. i guess my review auto-removed the gatekeeper request.
this is probably a good example of why you should not request gatekeeper before you get 2 +1s.

**Pull request merged**
